crates/hulk/build.rs

```diff
@@ -0,0 +1,118 @@
+                    "control::center_of_mass_provider",
+                    "control::fall_state_estimation",
+                    "control::game_controller_filter",
+                    "control::game_state_filter",
+                    "control::ground_contact_detector",
+                    "control::ground_provider",
+                    "control::kick_selector",
+                    "control::kinematics_provider",
+                    "control::led_status",
+                    "control::localization",
+                    "control::motion::arms_up_squat",
+                    "control::motion::condition_input_provider",
+                    "control::motion::dispatching_interpolator",
+                    "control::motion::energy_saving_stand",
+                    "control::motion::fall_protector",
+                    "control::motion::head_motion",
+                    "control::motion::joint_command_sender",
+                    "control::motion::jump_left",
+                    "control::motion::jump_right",
+                    "control::motion::look_around",
+                    "control::motion::look_at",
+                    "control::motion::motion_selector",
+                    "control::motion::sit_down",
+                    "control::motion::stand_up_back",
+                    "control::motion::stand_up_front",
+                    "control::motion::step_planner",
+                    "control::motion::walk_manager",
+                    "control::motion::walking_engine",
+                    "control::obstacle_filter",
+                    "control::odometry",
+                    "control::orientation_filter",
+                    "control::penalty_shot_direction_estimation",
+                    "control::primary_state_filter",
+                    "control::role_assignment",
+                    "control::rule_obstacle_composer",
+                    "control::sole_pressure_filter",
+                    "control::sonar_filter",
+                    "control::support_foot_estimation",
+                    "control::whistle_filter",
+                    "control::world_state_composer",
+                ],
+            },
+            CyclerManifest {
+                name: "SplNetwork",
+                kind: CyclerKind::Perception,
+                instances: vec![""],
+                setup_nodes: vec!["spl_network::message_receiver"],
+                nodes: vec![],
+            },
+            CyclerManifest {
+                name: "Audio",
+                kind: CyclerKind::Perception,
+                instances: vec![""],
+                setup_nodes: vec!["audio::microphone_recorder"],
+                nodes: vec!["audio::whistle_detection"],
+            },
+        ],
+    };
+    let root = "..";
+
+    let mut cyclers = Cyclers::try_from_manifest(manifest, root)?;
+    for path in cyclers.watch_paths() {
+        println!("cargo:rerun-if-changed={}", path.display());
```

Comment:
Recheck why the binary rebuilds every time.

Reply:
This was indeed caused by the `rerun-if-changed` print for the missing framework.toml.
Now that that's removed it nothing has to rebuilt if no files changed and pepsi build finishes within a few milliseconds.